python/cog/server/scope.py

```diff
@@ -0,0 +1,31 @@
+from contextlib import contextmanager
+from contextvars import ContextVar
+from typing import Callable, Generator, Optional, Union
+
+
+class Scope:
+    def __init__(
+        self,
+        *,
+        record_metric: Callable[[str, Union[float, int]], None],
+    ) -> None:
+        self.record_metric = record_metric
+
+
+SCOPE: ContextVar[Optional[Scope]] = ContextVar("scope", default=None)
```

Comment:
Very much a nit, but I'd call this `_current_scope` or similar. It's not a "constant".